packages/@uppy/core/src/Uppy.js

```diff
@@ -547,12 +547,12 @@ class Uppy {
           this.log(`Replaced the blob in the restored ghost file: ${newFile.name}, ${newFile.id}`)
         }
 
-        if (this.checkIfFileAlreadyExists(newFile.id)) {
+        const onBeforeFileAddedResult = this.opts.onBeforeFileAdded(newFile, nextFilesState)
+
+        if (!onBeforeFileAddedResult && this.checkIfFileAlreadyExists(newFile.id)) {
```

Comment:
```suggestion
        if (onBeforeFileAddedResult === false && this.checkIfFileAlreadyExists(newFile.id)) {
```

shoudln't we check for `=== false` here also like we do below?

Reply:
it's the same thing in this case

Reply:
what's the purpose of this  `=== false` check? couldn't it have also just said `!onBeforeFileAddedResult`?
https://github.com/transloadit/uppy/blob/5e522ee84e4b979fb76fb5c0ecf76c4a2b83e75e/packages/%40uppy/core/src/Uppy.js#L556

Reply:
Because we also allow it to override the file if you return an object.